hie-plugin-api/Haskell/Ide/Engine/Config.hs

```diff
@@ -54,7 +54,7 @@ instance Default Config where
 -- TODO: Add API for plugins to expose their own LSP config options
 instance FromJSON Config where
   parseJSON = withObject "Config" $ \v -> do
-    s <- v .: "languageServerHaskell"
+    s <- v .: "haskell"
```

Comment:
I think we need to accept both versions initially, here and in hls, otherwise we are forcing every client to make the change immediately, and banning them from running older versions of hie/hls.